clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -2283,6 +2307,37 @@ public void testRestOffsetsAuthorizationFailure() {
         assertEquals(5, subscriptions.position(tp0).offset);
     }
 
+    @Test
+    public void testPendingRevacationPartitionFetching() {
```

Comment:
nit: Revocation is misspelled

I did not find the name very clear. It looks like the main difference between this and `testFetchingPendingPartitions` is that this method tests that the pending state gets reset after reassignment? Perhaps the name should reflect that?